src/components/WorkerTypesTable/index.jsx

```diff
@@ -0,0 +1,214 @@
+import { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { withStyles } from 'material-ui/styles';
+import { TableCell, TableRow } from 'material-ui/Table';
+import { ListItemText } from 'material-ui/List';
+import Typography from 'material-ui/Typography';
+import ContentCopyIcon from 'mdi-react/ContentCopyIcon';
+import InformationVariantIcon from 'mdi-react/InformationVariantIcon';
+import { string, func, array, shape, arrayOf } from 'prop-types';
+import { find, propEq, memoizeWith, toString } from 'ramda';
+import { camelCase } from 'change-case';
+import LinkIcon from 'mdi-react/LinkIcon';
+import ButtonDrawer from '../ButtonDrawer';
+import StatusLabel from '../StatusLabel';
+import DateDistance from '../DateDistance';
+import TableCellListItem from '../TableCellListItem';
+import WorkerTypeMetadataCard from '../WorkerTypeMetadataCard';
+import ConnectionDataTable from '../ConnectionDataTable';
+import { VIEW_WORKER_TYPES_PAGE_SIZE } from '../../utils/constants';
+import sort from '../../utils/sort';
+import {
+  pageInfo,
+  awsProvisionerWorkerTypeSummary,
+} from '../../utils/prop-types';
+
+@withStyles(theme => ({
+  infoButton: {
+    marginLeft: -theme.spacing.double,
+    marginRight: theme.spacing.unit,
+  },
+}))
+/**
+ * Display relevant information about worker types in a table.
+ */
+export default class WorkerTypesTable extends Component {
+  static propTypes = {
+    /** Provisioner identifier */
+    provisionerId: string.isRequired,
+    /** Callback function fired when a page is changed. */
+    onPageChange: func.isRequired,
+    /** Worker Types GraphQL PageConnection instance. */
+    workerTypesConnection: shape({
+      edges: array,
+      pageInfo,
+    }).isRequired,
+    /**
+     * AWS worker-type summaries.
+     * Required when `provisionerId === 'aws-provisioner-v1'`.
+     */
+    awsProvisionerWorkerTypeSummaries: arrayOf(awsProvisionerWorkerTypeSummary),
+  };
+
+  static defaultProps = {
+    awsProvisionerWorkerTypeSummaries: null,
+  };
+
+  state = {
+    sortBy: null,
+    sortDirection: null,
+  };
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  createSortedWorkerTypesConnection = memoizeWith(
+    () =>
+      `${toString(this.props.workerTypesConnection.edges)}-${
+        this.state.sortBy
+      }-${this.state.sortDirection}`,
+    () => {
+      const { sortBy, sortDirection } = this.state;
+      const {
+        workerTypesConnection,
+        awsProvisionerWorkerTypeSummaries,
+      } = this.props;
+      const sortByProperty = camelCase(sortBy);
+      // Normalize worker types for aws-provisioner-v1
```

Comment:
Do you think this normalization should happen on the server, or in a util function?

Reply:
I think a util function makes more sense here. We have 14 provisioners and this normalization is only for `aws-provisioner`.